tests/calculations/test_autoinvalidate_cache.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""Test the automatic 'invalidates_cache' attribute for exit codes."""
+
+from __future__ import absolute_import
+import pytest
+
+import aiida
+from aiida.plugins import CalculationFactory
+from aiida.plugins.entry_point import get_entry_point_names
+
+if aiida.__version__ < '1.1':
+    pytest.skip("The 'invalidates_cache' feature is only available on AiiDA 1.1+", allow_module_level=True)
+
+QE_CALC_ENTRY_POINT_NAMES = [
+    ep_name for ep_name in get_entry_point_names(group='aiida.calculations') if ep_name.startswith('quantumespresso')
+]
+
+# When explicitly overriden 'invalidates_cache' are added, add entries
+# EXPLICIT_OVERRIDES = {<entry_point_name>: {<status_integer>: <expected_value>}}
+
+EXPLICIT_OVERRIDES = {}
+
+
+@pytest.mark.parametrize('entry_point_name', QE_CALC_ENTRY_POINT_NAMES)
+def test_exit_code_invalidates_cache(entry_point_name):
+    """Test automatic 'invalidates_cache' attribute of exit codes.
+
+    Test that the 'invalidates_cache' attribute of exit codes is automatically
+    set according to the status integer.
+    """
+    calc_class = CalculationFactory(entry_point_name)
+    overrides = EXPLICIT_OVERRIDES.get(entry_point_name, {})
+    for exit_code in calc_class.exit_codes.values():
+        if exit_code.status in overrides:
+            assert exit_code.invalidates_cache == overrides[exit_code.status]
```

Comment:
What is the added value of also checking the value of `invalidates_cache` for the overrides? Since by adding an exit status to the override you are basically saying that it doesn't follow the normal convention, what actual value is set doesn't really matter no? Wouldn't a list of excluded exit statuses not be sufficient?

Reply:
Yeah, I guess that would be sufficient.